common/etc/nginx/templates/default.conf.template

```diff
@@ -19,11 +19,10 @@ map $uri_full_path $uri_path {
     default $PREFIX_LEADING_DIRECTORY_PATH$uri_full_path;
 }
 
-map $S3_STYLE $s3_host_hdr {
-    virtual "${S3_BUCKET_NAME}.${S3_SERVER}";
-    path    "${S3_SERVER}:${S3_SERVER_PORT}";
-    default "${S3_BUCKET_NAME}.${S3_SERVER}";
-}
+# FINAL_S3_SERVER is set in the startup script
+# (either ./common/docker-entrypoint.sh or ./standalone_ubuntu_oss_install.sh)
+# based on the S3_STYLE configuration option.
+js_var $s3_host ${FINAL_S3_SERVER};
```

Comment:
**Annotation:**
This is the primary change.  We determine this value in the startup script then template it in before NGINX starts.  While before this value was calculated in three different places, using `js_var` allows access to its value both in the njs script and in the configuration.